### PR TITLE
ONNX: fix bug in export of ops involving torch.bool type

### PR DESCRIPTION
**Pull request by yaeldekel**
When an op involves creating a tensor of a certain type (such as torch.ones(...)), the tracer creates a `prim::Constant` node with an integer value representing the type. The mapping from the torch type to integers maps:
```
torch.complex32 -> 8
torch.complex64 -> 9
torch.complex128 -> 10
torch.bool -> 11
```
However, when the ONNX exporter maps back the integer to torch type, 10 is mapped to bool, 9 is mapped to complex128 and 8 is mapped to complex64.

### CODE REVIEW

**Comment by take-cheeze** [comment]
Seems to solve https://github.com/pytorch/pytorch/issues/32280

**Review by neginraoof — APPROVED** [review]
Looks good. Thanks!

**Comment by KsenijaS** [comment]
looks good. LGTM!

**Comment by vadimkantorov** [comment]
Currently `torch.ones(2, 4, dtype=torch.bool).sum(dim = 1)` failed for me (and `.to(torch.int32)` prior to `sum(...)` helped). Will this PR fix it as well?

**Review by houseroad — APPROVED** [review]
Thanks for the fix.

**Comment by twangnh** [comment]
I'm wondering which version of pytorch should I install for having this fix ?
